index.html

```diff
@@ -0,0 +1,191 @@
+        </pre>
+        <p>
+          User agents that do not support direct sockets SHOULD NOT expose
+          {{Navigator/openTCPSocket()}} or {{Navigator/openUDPSocket()}} on the
+          {{Navigator}} interface.
+        </p>
+        <section data-dfn-for="SocketOptions">
+          <h3>
+            `SocketOptions` dictionary
+          </h3>
+          <pre class="idl">
+          dictionary SocketOptions {
+            DOMString localAddress;
+            unsigned short localPort;
+
+            DOMString remoteAddress;
+            unsigned short remotePort;
+
+            unsigned long sendBufferSize;
+            unsigned long receiveBufferSize;
+
+            boolean keepAlive;
+            boolean noDelay;
+          };
+        </pre>
+          <p>
+            The <dfn>SocketOptions</dfn> dictionary consists of several
+            optional members:
+          </p>
+          <dl data-sort="">
+            <dt>
+              <dfn>localAddress</dfn> member
+            </dt>
+            <dt>
+              <dfn>localPort</dfn> member
+            </dt>
+            <dt>
+              <dfn>remoteAddress</dfn> member
+            </dt>
+            <dt>
+              <dfn>remotePort</dfn> member
+            </dt>
+            <dt>
+              <dfn>sendBufferSize</dfn> member
+            </dt>
+            <dt>
+              <dfn>receiveBufferSize</dfn> member
+            </dt>
+            <dt>
+              <dfn>keepAlive</dfn> member
+            </dt>
+            <dd>
+              Only relevant for TCP.
+            </dd>
+            <dt>
+              <dfn>noDelay</dfn> member
+            </dt>
+            <dd>
+              Only relevant for TCP.
+            </dd>
+          </dl>
+          <div class="note">
+            The only current implementation (Chromium) ignores
+            {{SocketOptions/localAddress}} and {{SocketOptions/localPort}}.
+          </div>
+        </section>
+        <section data-dfn-for="TCPSocket">
+          <h3>
+            `TCPSocket` interface
+          </h3>
+          <pre class="idl">
+          [Exposed=Window, SecureContext] interface TCPSocket {
+            Promise&lt;undefined&gt; close();
+
+            readonly attribute DOMString remoteAddress;
+            readonly attribute unsigned short remotePort;
+            readonly attribute ReadableStream readable;
+            readonly attribute WritableStream writable;
+          };
+        </pre>
+          <div class="note">
+            The only current implementation (Chromium) does not support "byob"
+            mode for {{TCPSocket/readable}}. See [[streams]] for more
+            information about streams.
+          </div>
+        </section>
+        <section data-dfn-for="UDPSocket">
+          <h3>
+            `UDPSocket` interface
+          </h3>
+          <pre class="idl">
+          [Exposed=Window, SecureContext] interface UDPSocket {
+            async iterable&lt;SocketOptions, Blob&gt;;
```

Comment:
Only remoteAddress and remotePort are relevant from SocketOptions. We could define a new dictionary.

They will be important if/when we add more general UDP support. Unix [recvfrom() / recvmsg()](https://linux.die.net/man/2/recvfrom) and Chromium [UDPSocketListener](https://source.chromium.org/chromium/chromium/src/+/master:services/network/public/mojom/udp_socket.mojom;drc=71251a724053fd101e07f664914f06b9b4372038;l=225) provide this information with each incoming datagram.

Reply:
Seems fine to use SocketOptions then. Maybe note here if particular fields are required/expected to be populated?

Reply:
done